simd/src/Kokkos_SIMD_AVX512.hpp

```diff
@@ -2484,18 +2614,200 @@ namespace Experimental {
+        other) noexcept
+    : m_value(_mm512_cvtepu64_pd(static_cast<__m512i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<double, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value(_mm512_cvtepi32_pd(static_cast<__m256i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<double, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::uint32_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value(_mm512_cvtepi32_pd(static_cast<__m256i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<double, simd_abi::avx512_fixed_size<8>> const& other) noexcept
+    : m_value(_mm512_cvtpd_ps(static_cast<__m512d>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::int64_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value(_mm512_cvtepi64_ps(static_cast<__m512i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::uint64_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value(_mm512_cvtepu64_ps(static_cast<__m512i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value(_mm256_cvtepi32_ps(static_cast<__m256i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::uint32_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value(_mm256_cvtepi32_ps(static_cast<__m256i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<16>>::basic_simd(
+    basic_simd<std::int32_t, simd_abi::avx512_fixed_size<16>> const&
+        other) noexcept
+    : m_value(_mm512_cvtepi32_ps(static_cast<__m512i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<float, simd_abi::avx512_fixed_size<16>>::basic_simd(
+    basic_simd<std::uint32_t, simd_abi::avx512_fixed_size<16>> const&
+        other) noexcept
+    : m_value(_mm512_cvtepi32_ps(static_cast<__m512i>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<std::int32_t, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<float, simd_abi::avx512_fixed_size<8>> const& other) noexcept
+    : m_value(_mm256_cvtps_epi32(static_cast<__m256>(other))) {}
+
+KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION
+basic_simd<std::int32_t, simd_abi::avx512_fixed_size<8>>::basic_simd(
+    basic_simd<std::uint32_t, simd_abi::avx512_fixed_size<8>> const&
+        other) noexcept
+    : m_value((static_cast<__m256i>(other))) {}
```

Comment:
The extra parentheses are unnecessary
```suggestion
    : m_value(static_cast<__m256i>(other)) {}
```

Reply:
Applied.